src/routes/api/auth/index.ts

```diff
@@ -24,22 +21,30 @@ const plugin: FastifyPluginAsyncTypebox = async (fastify) => {
     async function (request, reply) {
       const { username, password } = request.body
 
-      const user = await fastify.repository.find<Auth>('users', {
-        select: 'username, password',
-        where: { username }
-      })
+      const user = await fastify.knex<Credentials>('users')
+        .select('username', 'password')
+        .where({ username })
+        .first()
 
       if (user) {
         const isPasswordValid = await fastify.compare(password, user.password)
         if (isPasswordValid) {
-          const token = fastify.jwt.sign({ username: user.username })
+          const roles = await fastify.knex<{ name: string }>('roles')
+            .select('roles.name')
+            .join('user_roles', 'roles.id', '=', 'user_roles.role_id')
+            .join('users', 'user_roles.user_id', '=', 'users.id')
+            .where('users.username', username)
+
+          const token = fastify.jwt.sign({
+            username: user.username,
+            roles: roles.map((role) => role.name)
+          })
```

Comment:
It does look more performant to put the roles in the jwt than doing an sql query at each request.
But it might have a drawback, e.g. I want a user not being an admin anymore, It could take some time to apply the changes (JWT expiration time).